chat-widget/src/plugins/createChatTranscript.ts

```diff
@@ -0,0 +1,395 @@
+                        bubbleTextColor: '${AgentFontColor}',
+                        bubbleBorderRadius: 12,
+                        bubbleNubSize: 1,
+                        bubbleFromUserBackground: '${CustomerDialogColor}',
+                        bubbleFromUserTextColor: '${CustomerFontColor}',
+                        bubbleFromUserBorderRadius: 12,
+                        bubbleFromUserNubSize: 1,
+                        botAvatarInitials: 'C1',
+                        userAvatarInitials: 'C2'
+                    };
+
+                    window.WebChat.renderWebChat({
+                        directLine: adapter,
+                        styleOptions,
+                        activityStatusMiddleware,
+                        avatarMiddleware
+                    },
+                    document.getElementById('transcript'));
+                <\/script>
+            </body>
+        `;
+
+        return htmlData;
+    }
+
+    createHTML() {
+        const htmlData = `
+            <html>
+                ${this.createHeadElement()}
+                ${this.createBodyElement()}
+            </html>
+        `;
+
+        return htmlData;
+    }
+}
+
+const download = (fileName: string, htmlData: string) => {
+    const aElement = document.createElement("a");
+
+    const blob = new Blob([htmlData], {type: "text/html"});
+    const objectUrl = URL.createObjectURL(blob);
+
+    aElement.setAttribute("href", objectUrl);
+    aElement.setAttribute("download", fileName);
+    aElement.style.display = "none";
+
+    document.body.appendChild(aElement);
+    aElement.click();
+    document.body.removeChild(aElement);
+};
+
+const createChatTranscript = async (transcript: string, chatSDK: any, renderAttachments = false) => { // eslint-disable-line @typescript-eslint/no-explicit-any
+    const transcriptMessages = JSON.parse(transcript);
+
+    const convertBlobToBase64 = async (blob: Blob) => {
+        return new Promise((resolve, _) => {
+            const reader = new FileReader();
+            reader.onloadend = () => resolve(reader.result);
+            reader.readAsDataURL(blob);
+        });
+    };
+
+    let messages = transcriptMessages;
+
+    if (renderAttachments) {
+        messages = await Promise.all(transcriptMessages.map(async (message: any) => { // eslint-disable-line @typescript-eslint/no-explicit-any
+            const {amsReferences, amsMetadata } = message;
+            if (amsReferences && amsMetadata) {
+                const references = JSON.parse(amsReferences);
+                const metadata = JSON.parse(amsMetadata);
+                const fileMetadata = {
+                    id: references[0],
+                    type: metadata[0].contentType
+                };
+
+                const blob = await chatSDK.downloadFileAttachment(fileMetadata);
+                const base64 = await convertBlobToBase64(blob);
+                message.contentUrl = base64;

```

Comment:
Same question here about file attachments being shown in the transcript -- should we be able to see attachment content aside from just text?

Reply:
Currently, we're only rendering a text messages when a file attachment was uploaded. It may be something we can support in the future, but it will require further investigation & design